Toolset/libraries/revideextensionlibrary.livecodescript

```diff
@@ -838,35 +862,82 @@ function revIDEExtensionsOrderByDependency pExtensions
    return __dependencyOrder(tDependencies, pExtensions)
 end revIDEExtensionsOrderByDependency
 
+command revIDEExtensionFetchSourceFromFolder pFolder, @rSource, @rType
+   local tFiles
+   put files(pFolder) into tFiles
+   
+   local tExtFile, tType
+   filter tFiles with "*.lcb" into tExtFile
+   put "lcb" into tType
+   if tExtFile is empty then
+      put "lcs" into tType
+      filter tFiles with regex pattern ".*\.livecode(script)?$" into tExtFile
+   end if
+
+   put tExtFile into rSource
+   put tType into rType
+end revIDEExtensionFetchSourceFromFolder
+
+private command __FindExtensionsInFolderRecursive pFolder, pIsUserFolder, @xDataA
+   local tExtSource, tArray, tFiles, tExtFile, tType
+   revIDEExtensionFetchSourceFromFolder pFolder, tExtFile, tType
+   
+   local tNumExtensions
+   put the number of lines in tExtFile into tNumExtensions
+   if tNumExtensions > 1 then
+      get __revIDEError("Invalid extension folder" && pFolder & ": require exactly one extension per folder")
+      exit __FindExtensionsInFolderRecursive
+   end if
+   
+   -- If we found no extensions, recurse
+   if tNumExtensions is 0 then
+      repeat for each line tSubFolder in folders(pFolder)
+         if tSubFolder begins with "." then next repeat
+         __FindExtensionsInFolderRecursive pFolder & slash & tSubFolder, pIsUserFolder, xDataA
+      end repeat
+      exit __FindExtensionsInFolderRecursive
+   end if
+   
+   put pFolder & slash & tExtFile into tExtSource
+   if tType is "lcb" then
+      put __fetchModuleData(tExtSource) into tArray
+   else
+      put __fetchScriptLibraryData(tExtSource) into tArray
+   end if
+   
+   put tExtFile into tArray["source_file"]
```

Comment:
I think `revIDEExtensionFetchSourceFromFolder` may feasibly set this to a list of files. Perhaps it should be an error if there is something other than a single file found in `revIDEExtensionFetchSourceFromFolder`?

Reply:
It does do that, above: https://github.com/livecode/livecode-ide/pull/1773/files/fca791decc95eb8c276ae94edbcf14b28baccc4c#diff-5a54288a32499468dd2cab9f178bed1dR891